modules/linear_boltzmann_solvers/lbs_solver/iterative_methods/classic_richardson.h

```diff
@@ -0,0 +1,65 @@
+// SPDX-FileCopyrightText: 2024 The OpenSn Authors <https://open-sn.github.io/opensn/>
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "modules/linear_boltzmann_solvers/lbs_solver/iterative_methods/wgs_context.h"
+#include "modules/linear_boltzmann_solvers/lbs_solver/groupset/lbs_groupset.h"
+#include "framework/math/linear_solver/linear_solver.h"
+#include <memory>
+#include <vector>
+#include <functional>
+
+namespace opensn
+{
+namespace lbs
+{
+
+/**Linear Solver specialization for Within GroupSet (WGS) solves with classic
+ * Richardson.
+ */
+class ClassicRichardson : public LinearSolver
+{
+public:
+  /**
+   * Constructor.
+   * \param gs_context_ptr Context Pointer to abstract context.
+   */
+  explicit ClassicRichardson(std::shared_ptr<WGSContext> gs_context_ptr);
+
+  ~ClassicRichardson() override;
+
+  void Setup() override {}
+
+  void Solve() override;
+
+protected:
+  void PreSetupCallback() override {}
+
+  void SetConvergenceTest() override {}
+
+  void SetSystemSize() override {}
+
+  void SetSystem() override {}
+
+  void SetPreconditioner() override {}
+
+  void PostSetupCallback() override {}
+
+  void PreSolveCallback() override {}
+
+  void SetRHS() override {}
+
+  void SetInitialGuess() override {}
+
+  void PostSolveCallback() override {}
+
+private:
+  std::vector<double> saved_q_moments_local_;
+  std::vector<double> psi_new_, psi_old_;
+
+  double ComputePointwisePsiChange();
```

Comment:
This feels like it should belong in `DiscreteOrdinatesSolver` to be honest. However, since this class does not necessarily have an instance of that (even through contexts), I am not sure how to make that work.

Reply:
I disagree. This is specific to this WGS solver and uses private member variables. I don't see that it will ever be used anywhere else.

Reply:
That is fair. The same argument be made for `ComputePointwisePhiChange` which was implemented in `LBSSolver`? Where should that live instead?

Reply:
Yes, the same argument could be made for `ComputePointwisePhiChange`. I'll make it a private method. The reason I originally had it in `LBSSovler` was that I expected the functionality to overlap with the pointwise phi check for the refactored AGS solver. But, it turned out that trying to reuse that code would result in a slew of unnecessary collective MPI operations.

Reply:
I won't hold things up on this point, but it could be worth considering defining these methods as standalone (i.e. taking an instance of `LBSSolver` and the new and old vectors) somewhere in the LBS module to avoid these issues, if that is feasible.